lib/librato/rack/tracker.rb

```diff
@@ -78,10 +77,10 @@ def should_start?
           log :debug, 'halting: credentials not present.'
         elsif config.autorun == false
           log :debug, 'halting: LIBRATO_AUTORUN disabled startup'
-        elsif qualified_source !~ SOURCE_REGEX
-          log :warn, "halting: '#{qualified_source}' is an invalid source name."
-        elsif on_heroku && !config.explicit_source?
-          log :warn, 'halting: source must be provided in configuration.'
+        elsif qualified_tags.any? { |k,v| k.to_s !~ ValidatingQueue::TAGS_KEY_REGEX || v.to_s !~ ValidatingQueue::TAGS_VALUE_REGEX }
+          log :warn, "halting: '#{qualified_tags}' are invalid tags."
+        elsif on_heroku && !config.has_tags?
+          log :warn, 'halting: tags must be provided in configuration.'
         else
```

Comment:
Yeah, we should probably figure out what the most tags we use on any specific instrumentation and set a hard limit of 4 or 5 here. This is another reason I favor dropping the pid one.